src/recovery.rs

```diff
@@ -1116,7 +1182,7 @@ mod tests {
             Ok(())
         );
 
-        assert_eq!(r.sent[packet::EPOCH_APPLICATION].len(), 1);
+        assert_eq!(r.sent[packet::EPOCH_APPLICATION].len(), 2);
```

Comment:
before calling on_ack_received() sent[] is [0, 1, 2, 3] (I only use packet number here, not ranges) and acked[] = [0, 1, 3]. Then after calling on_ack_received(), sent[] should be [2] I think? But when I checked sent[] at this line it contains [2, 3].

Reply:
That's on purpose. From the commit message/PR description:
> At predefined times, the semt packets marked as acked or lost are then
removed from the start of the queue in contiguous blocks, in order to
avoid having to copy elements around to cover gaps in the queue caused
by removed elements.

Reply:
That is, we only remove packets from the start of the list, so if there are gaps like in this case the packets are not actually removed, but are still marked as lost or acked so they are ignored.

Reply:
Ok. It would be great to add some comment there, not to be confused later. Thanks!